lib/gameState/CGameState.h

```diff
@@ -39,12 +39,67 @@ struct SThievesGuildInfo;
 class CRandomGenerator;
 class GameSettings;
 
-struct UpgradeInfo
+class UpgradeInfo
```

Comment:
Since this is no longer tiny struct, can you move it to a separate file in same directory?

Reply:
fields haven't changed, just moved 2 of them to private and added few functions. Should I make it back a struct or stay with class and move it to its own header and source files ?

Reply:
Move to own header (and source file if necessary).

By "tiny struct" I mean not size in memory and such, but rather size of code, which is now rather large due to having ~10 new methods.